drivers/gpio/gpio_numaker.c

```diff
@@ -226,6 +227,35 @@ static void gpio_numaker_isr(const struct device *dev)
 	gpio_fire_callbacks(&data->callbacks, dev, int_status);
 }
 
+#ifdef CONFIG_CLOCK_CONTROL_NUMAKER_PCC
+#define NUMAKER_DT_INST_CLOCK_INIT(n)                           \
+    .clk_dev    = DEVICE_DT_GET(DT_INST_CLOCKS_CTLR(n)),        \
+    .clk_modidx = DT_INST_CLOCKS_CELL(n, clock_module_index),
+#else
+#define NUMAKER_DT_INST_CLOCK_INIT(n)                           \
+    .clk_modidx = DT_INST_CLOCKS_CELL(n, clock_module_index),
+#endif
+
+#ifdef CONFIG_CLOCK_CONTROL_NUMAKER_PCC
+#define GPIO_NUMAKER_CLOCK_INIT(n)                  \
+    do {                                            \
+        const struct gpio_numaker_config *config = dev->config;                 \
+        int err = clock_control_on(config->clk_dev, \
+                               (clock_control_subsys_t) config->clk_modidx);    \
+        if (err != 0) {                             \
+            return err;                             \
+        }                                           \
+    } while (0)
+#else
+#define GPIO_NUMAKER_CLOCK_INIT(n)                  \
+    do {                                            \
+        const struct gpio_numaker_config *config = dev->config;                 \
+        SYS_UnlockReg();                            \
+        CLK_EnableModuleClock(config->clk_modidx);  \
+        SYS_LockReg();                              \
+    } while (0)
+#endif
```

Comment:
I not prefer this way.
It's too complex for simple CLK init procedure.
To store index/info in config structure, let gpio_numaker_configure() handle it instead of macro, like as 
#ifdef CONFIG_CLOCK_CONTROL_NUMAKER_PCC

#else
#endif